junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -28,6 +29,8 @@
     private final CountDownLatch startable;
     private final CountDownLatch finished;
     private volatile Thread testThread;
+    
+    private final AtomicBoolean exceptionOccurred = new AtomicBoolean(false); 
```

Comment:
Actually, I don't get the one and only call to `setInterruptOnException()`, in [SpecificationStatement](https://github.com/k3po/k3po/blob/7ab1dd496cda4bfb5b9d2659590746ba8d237967/junit/src/main/java/org/kaazing/k3po/junit/rules/SpecificationStatement.java).  It is 

```java
      latch.setInterruptOnException(Thread.currentThread());
```

Meaning the thread is interrupting itself, which I believe is a NOOP.  Thus we can refactor `setInterruptOnException` to something like
```
     public void setTestThread(Thread testThread) {
          this.testThread = testThread
      } 
```

This should completely remove the need for the Atomic.